feat: Gru HTTP

sdelamo opened the pull request:
See: https://agorapulse.github.io/gru/

Adds feature for GRU - HTTP interaction testing framework with out-of-box support for Micronaut.

sdelamo committed:
feat: gru-http

See: https://agorapulse.github.io/gru/

sdelamo commented:
@musketyr check it this PR and let me know if you have any concerns.

musketyr commented on starter-core/src/main/java/io/micronaut/starter/feature/agora/AgoraFeature.java:
```diff
@@ -0,0 +1,31 @@
+
+import io.micronaut.starter.feature.Feature;
+
+public interface AgoraFeature extends Feature {
```
just to keep up with branding, can we go with full `AgorapulseFeature` and `agorapulse` package? Thank you

sdelamo replied:
done

musketyr commented on starter-core/src/main/java/io/micronaut/starter/feature/agora/gru/template/helloWorldGruControllerTestKotlin.rocker.raw:
```diff
@@ -0,0 +1,35 @@
+    @@Test
+    @@Throws(Throwable::class)
+    fun testGet() {
+        gru.verify { test: TestDefinitionBuilder ->
+            test["/gru"]
+                .expect { response: ResponseDefinitionBuilder ->
+                    response.json(
+                        "gruIndex.json"
+                    )
+                }
+        }
+    }
```
This can be simplified as 

```suggestion
        gru.verify { it
            .get("/gru")
            .expect { it.json("gruIndex.json") }
        }
    }
```

musketyr commented on starter-core/src/main/resources/pom.xml:
```diff
@@ -12,6 +12,11 @@
+        <dependency>
+            <groupId>com.agorapulse</groupId>
+            <artifactId>gru-micronaut</artifactId>
+            <version>1.0.1</version>
```
I have release some minor patches this morning. one of them upgrading JSoup so Gru is no longer highlighted by IntelliJ due security vulnerability.

```suggestion
            <version>1.0.3</version>
```

musketyr reviewed (commented):
Thank you very much for this PR! I have added some minor suggestions in individual comments. Can we also support Kotest? I was able to create a simple test

```
import com.agorapulse.gru.Gru
import io.micronaut.test.extensions.kotest.annotation.MicronautTest
import io.kotest.core.spec.style.StringSpec

@MicronautTest
class HelloWorldGruControllerTest(private val gru: Gru): StringSpec({

    "get hello message" {
        gru.verify { it
            .get("/gru")
            .expect { it.json("gruIndex.json") }
        }
    }

})
```

sdelamo committed:
gru to 1.0.3

Co-authored-by: Vladimir Orany <vladimir@orany.cz>

sdelamo committed:
simplify gru kotlin test

Co-authored-by: Vladimir Orany <vladimir@orany.cz>

sdelamo committed:
move to agorapulse and add kotest

sdelamo committed:
test is community

sdelamo commented:
> Thank you very much for this PR! I have added some minor suggestions in individual comments. Can we also support Kotest? I was able to create a simple test
> 
> ```
> import com.agorapulse.gru.Gru
> import io.micronaut.test.extensions.kotest.annotation.MicronautTest
> import io.kotest.core.spec.style.StringSpec
> 
> @MicronautTest
> class HelloWorldGruControllerTest(private val gru: Gru): StringSpec({
> 
>     "get hello message" {
>         gru.verify { it
>             .get("/gru")
>             .expect { it.json("gruIndex.json") }
>         }
>     }
> 
> })
> ```

I have added Kotest

musketyr reviewed (approved):
🎉

sdelamo committed:
use default test framework for lang
